torchmultimodal/models/gpt.py

```diff
@@ -633,16 +633,17 @@ def forward(
         )
 
         if return_attn_weights:
-            x, attn_probs = attn_out
+            hidden, attn_probs = attn_out
         else:
-            x = attn_out
+            hidden = attn_out
 
         if use_cache:
             past_key_values = self.attention.cache
 
-        x = x + self.dropout_attn(x)
-        x = self._mlp_block(self.norm_mlp(x))
-        x = x + self.dropout_mlp(x)
+        x = x + self.dropout_attn(hidden)
+
+        hidden = self._mlp_block(self.norm_mlp(x))
```

Comment:
nit: I think the use of `hidden` here is kinda confusing. Maybe we can more closely follow naming in the [existing transformer layer](https://github.com/facebookresearch/multimodal/blob/main/torchmultimodal/modules/layers/transformer.py#L120-L168)?

Reply:
Here I am also exposing the option to return the attention probabilities so it's not as straightforward to chain the outputs like in what you quoted (where `return_attn_weights` is always `False`, [ref](https://github.com/facebookresearch/multimodal/blob/main/torchmultimodal/modules/layers/transformer.py#L95)).

I am renaming `hidden` to `attn_hidden_states` and `mlp_hidden_states` to be clear.